Assets/MRTK/Tests/TestUtilities/TestMotionController.cs

```diff
@@ -0,0 +1,147 @@
+            if (waitForFixedUpdate)
+            {
+                yield return new WaitForFixedUpdate();
+            }
+        }
+
+        /// <inheritdoc />
+        public override IEnumerator Hide(bool waitForFixedUpdate = true)
+        {
+            yield return PlayModeTestUtilities.HideController(handedness, simulationService);
+            if (waitForFixedUpdate)
+            {
+                yield return new WaitForFixedUpdate();
+            }
+        }
+
+        /// <inheritdoc />
+        public override IEnumerator MoveTo(Vector3 newPosition, int numSteps = PlayModeTestUtilities.ControllerMoveStepsSentinelValue, bool waitForFixedUpdate = true)
+        {
+            Vector3 oldPosition = position;
+            position = newPosition;
+            for (var iter = PlayModeTestUtilities.MoveMotionController(oldPosition, newPosition, isSelecting, isGrabbing, isPressingMenu, handedness, simulationService, numSteps); iter.MoveNext();)
+            {
+                yield return iter.Current;
+            }
+            if (waitForFixedUpdate)
+            {
+                yield return new WaitForFixedUpdate();
+            }
+        }
+
+        /// <inheritdoc />
+        public override IEnumerator Move(Vector3 delta, int numSteps = PlayModeTestUtilities.ControllerMoveStepsSentinelValue)
+        {
+            for (var iter = MoveTo(position + delta, PlayModeTestUtilities.CalculateNumSteps(numSteps)); iter.MoveNext();)
+            {
+                yield return iter.Current;
+            }
+        }
+
+        /// <inheritdoc />
+        public override IEnumerator SetRotation(
+            Quaternion newRotation,
+            int numSteps = PlayModeTestUtilities.ControllerMoveStepsSentinelValue)
+        {
+            Quaternion oldRotation = rotation;
+            rotation = newRotation;
+            yield return PlayModeTestUtilities.SetMotionControllerRotation(
+                oldRotation,
+                newRotation,
+                position,
+                isSelecting,
+                isGrabbing,
+                isPressingMenu,
+                handedness,
+                PlayModeTestUtilities.CalculateNumSteps(numSteps),
+                simulationService);
+        }
+
+        /// <summary>
+        /// Changes the state of the simulated motion controller.
+        /// </summary>
+        /// <param name="isSelecting">Whether the motion controller should be selecting something.</param>
+        /// <param name="isGrabbing">Whether the motion controller should be grabbing something.</param>
+        /// <param name="isPressingMenu">Whether the menu button of the motion controller should be pressed down.</param>
+        /// <param name="waitForFixedUpdate">If true, waits for a fixed update after moving to the new state.</param>
+        public IEnumerator SetState(bool isSelecting, bool isGrabbing, bool isPressingMenu, bool waitForFixedUpdate = true)
```

Comment:
In general, functions that take a bunch of random bools are extremely hard to use.

For example, consider a caller:

SetState(true, false, false);

Now imagine that you went away for a week and came back - you pretty much MUST read the function definition to understand the code - i.e. it's not understandable by itself.

Usually for these cases, you'll want to have flags/enums or other named things to do stuff here like:

SetState(MotionControllerState.IsSelecting | MotionControllerState.IsPressingMenu);
SetState(MotionControllerState.IsGrabbing);

This is extremely clear from reading what the state is being set

Reply:
I agree that these functions tend to be difficult to use and understand without reading the documentation. I am wondering if there is a way to make it more readable while maintaining the flexibility here (i.e. the user can set the states of the three buttons independently of each other). I thought about enums but was concerned about making things too verbose (e.g. imagine `MotionControllerState.IsSelectingAndIsGrabbingAndIsPressingMenu`) Maybe a struct is the way to go?